### PR TITLE
Add phone hash label

### PR DESCRIPTION
**Pull request by tinakurian**
## Description
adding phone hash for banned user and removing phone number. 

## Checks
1. Have you run `make generate` target? **[yes/no]**
yes
2. Does `make generate` change anything in other projects (host-operator, member-operator, toolchain-common)? **[yes/no]** 
(NOTE: You can ignore it if the only changes are in the annotation `alm-examples` of the `ClusterServiceVersion` object)
no

Related to PR: https://github.com/codeready-toolchain/registration-service/pull/126

### CODE REVIEW

File: pkg/apis/toolchain/v1alpha1/usersignup_types.go [hunk 1]
```diff
@@ -14,8 +14,6 @@ const (
 
 	// UserSignupUserEmailAnnotationKey is used for the usersignup email annotations key
 	UserSignupUserEmailAnnotationKey = LabelKeyPrefix + "user-email"
-	// UserSignupPhoneNumberLabelKey is used for the usersignup phone number label key
-	UserSignupPhoneNumberLabelKey = LabelKeyPrefix + "phone-number"
```
Comment (alexeykazakov) [inline pkg/apis/toolchain/v1alpha1/usersignup_types.go:18]: We still need a label for the hash in usersignup.

Comment (tinakurian) [reply pkg/apis/toolchain/v1alpha1/usersignup_types.go:18]: thanks @alexeykazakov, i think we already have one here: https://github.com/codeready-toolchain/api/blob/master/pkg/apis/toolchain/v1alpha1/usersignup_types.go#L34-L35 so was just removing this one

Comment (alexeykazakov) [reply pkg/apis/toolchain/v1alpha1/usersignup_types.go:18]: Ah OK. Didn't see it. Thanks.

**Review by alexeykazakov — APPROVED** [review]

File: pkg/apis/toolchain/v1alpha1/banneduser_types.go [hunk 2]
```diff
@@ -8,8 +8,8 @@ const (
 	// BannedUserEmailHashLabelKey is used for the banneduser email hash label key
 	BannedUserEmailHashLabelKey = LabelKeyPrefix + "email-hash"
 
-	// BannedUserPhoneNumberLabelKey is used for the banneduser phone number label key
-	BannedUserPhoneNumberLabelKey = LabelKeyPrefix + "phone-number"
+	// BannedUserPhoneNumberHashLabelKey is used for the banneduser phone number label key
```
Comment (MatousJobanek) [inline pkg/apis/toolchain/v1alpha1/banneduser_types.go:11]: Just a detail
```suggestion
	// BannedUserPhoneNumberHashLabelKey is used a label key for the hash of a phone of the banned user
```

**Review by MatousJobanek — APPROVED** [review]

File: pkg/apis/toolchain/v1alpha1/banneduser_types.go (hunk 2 above)
Comment (tinakurian) [reply pkg/apis/toolchain/v1alpha1/banneduser_types.go]: done, thanks Matous!

**Pull request merged**